### ISSUE TITLE
Improve ARM Processor CPU Speed Detection

### ISSUE THREAD

**Issue by I-Rinka** [rcmaehl/WhyNotWin11#50]
This software is good but it only works on x86 processors. Even though it can run on WOA because of the compatibility layer, it can't provide comprehensive information. The WOA ecosystem is growing, so native ARM64 version may be needed.

![image](https://user-images.githubusercontent.com/50841088/123528834-8a585400-d71d-11eb-9398-15ad82c43eb6.png)

The supported Qualcomm processors are shown below.

[https://docs.microsoft.com/en-us/windows-hardware/design/minimum/supported/windows-11-supported-qualcomm-processors](url)

**Comment by micwoj92** [comment]
I don't think that native is possible, because autoit does not support support arm.

**Comment by rcmaehl** [comment]
I won't be able to make a native ARM64 version, but I can implement improvements for proper detection on Supported Qualcomm Processors. I'm assuming this is in regards to the Architecture check. Odd that it wasn't able to mark the CPU as Compatible as it's in the Qualcomm list. Finally, can you send me a copy of `Save All Information` from dxdiag?

**Comment by I-Rinka** [comment]
Sure : ) But my Windows is in Chinese. Hoping it would not trouble you.

[DxDiag.txt](https://github.com/rcmaehl/WhyNotWin11/files/6720863/DxDiag.txt)

> I won't be able to make a native ARM64 version, but I can implement improvements for proper detection on Supported Qualcomm Processors. I'm assuming this is in regards to the Architecture check. Odd that it wasn't able to mark the CPU as Compatible as it's in the Qualcomm list. Finally, can you send me a copy of `Save All Information` from dxdiag?

**Comment by rcmaehl** [comment]
`Driver Model: 2.6 WDDM`

DXDIAG WHYYYYY. Will fix this in an upcoming build to pass WDDM check

**Comment by micwoj92** [comment]
@I-Rinka Could you test latest stable release, are there any improvements?

**Comment by I-Rinka** [comment]
![image](https://user-images.githubusercontent.com/50841088/124114411-1ca28400-da9f-11eb-8eab-b29c57fbb795.png)
The left one is x64 version and the right is x86 version. Only CPU frequency has problem.

**Comment by micwoj92** [comment]
So a regression then :/

**Comment by rcmaehl** [comment]
> ![image](https://user-images.githubusercontent.com/50841088/124114411-1ca28400-da9f-11eb-8eab-b29c57fbb795.png)
> The left one is x64 version and the right is x86 version. Only CPU frequency has problem.

This is probably the same issue that intel chips have where it reports current speed as max speed instead of max possible speed.

**Comment by micwoj92** [comment]
@rcmaehl is the cpu speed exact same issue as with intel x86_64 cpus? if yes then this issue should be closed, because the applications works/detects same on arm and on x86_64

**Comment by rcmaehl** [comment]
@I-Rinka can you try the latest build:

https://nightly.link/rcmaehl/WhyNotWin11/workflows/wnw11/main/WNW11.zip

**Comment by I-Rinka** [comment]
![image](https://user-images.githubusercontent.com/50841088/135249113-2eb16d63-b309-49a1-8bcc-eb3e8b7c894f.png)

It works.

**Comment by I-Rinka** [comment]
Oh, the CPU Frequency is still not a correct value when I look close to it. It just turns green.

**Comment by rcmaehl** [comment]
Should be fixed in the latest build. 

https://nightly.link/rcmaehl/WhyNotWin11/workflows/wnw11/main/WNW11.zip

**Comment by I-Rinka** [comment]
![image](https://user-images.githubusercontent.com/50841088/135273051-a6023edb-ceab-4573-91ec-e5ca0c251edd.png)

Basically it's right. But.. I don't know, probably the emulated processor has a slightly lower frequency.

**Comment by rcmaehl** [comment]
Not sure how to improve this unfortunately, just glad it's properly passing now.

**Comment by WilloIzCitron** [comment]
th CPU Speed detects on registry only and it may inaccurate
